### PR TITLE
manifest: Update zephyr with doc tool versions

### PR DESCRIPTION
**Pull request by thst-nordic**
https://github.com/zephyrproject-rtos/zephyr/pull/48947
https://github.com/nrfconnect/sdk-zephyr/pull/910
Update to Sphinx 5.x, also update breathe since version 4.34 is the
first to officially support Sphinx 5.x.

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 44ede1e459c061fb1f7b65431535d6b66654ec78
+      revision: c637477892afef5e2186fe63463703aaab9c53b3
```
Comment (rlubos) [inline west.yml:53]: You should use `pull/910/head` at this point.

Comment (thst-nordic) [reply west.yml]: done

**Comment by thst-nordic** [comment]
waiting for +1 from @gmarull

File: west.yml [hunk 2]
```diff
@@ -50,7 +50,8 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 44ede1e459c061fb1f7b65431535d6b66654ec78
+      revision: pull/910/head
+      # revision: c637477892afef5e2186fe63463703aaab9c53b3
```
Comment (tejlmand) [inline west.yml:54]: be very careful here, cause you don't know what the SHA will be when https://github.com/nrfconnect/sdk-zephyr/pull/910 is merged.
but this is a real SHA and could (until garbage collected) be fetched by `west`, and therefor easily make it into `sdk-nrf/main`.

That is a risk I don't like, so I suggest to remove this line.

File: west.yml [hunk 3]
```diff
@@ -50,7 +50,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 44ede1e459c061fb1f7b65431535d6b66654ec78
+      revision: 85445474f29eed2ad09062456aaeb1dfc204e8cf
```
Comment (rlubos) [inline west.yml:53]: This should be `86893246053c50d34618f09ec6722cae8ba19472` (https://github.com/nrfconnect/sdk-zephyr/commit/86893246053c50d34618f09ec6722cae8ba19472)

**Review by gmarull — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**